src/drivers/dw/dma.c

```diff
@@ -425,17 +422,25 @@ static int dw_dma_stop(struct dma_chan_data *channel)
 	 */
 	dma_reg_write(dma, DW_CFG_LOW(channel->index),
 		      dw_chan->cfg_lo | DW_CFGL_SUSPEND | DW_CFGL_DRAIN);
+#else
+	dma_reg_write(dma, DW_CFG_LOW(channel->index),
+		      dw_chan->cfg_lo | DW_CFGL_SUSPEND);
+#endif
+
+	/* calculate different timeout for connection with host or dev */
+	copy_with_dev = channel->direction == DMA_DIR_MEM_TO_DEV;
+	copy_with_dev |= channel->direction == DMA_DIR_DEV_TO_MEM;
+	copy_with_dev |= channel->direction == DMA_DIR_DEV_TO_DEV;
+	timeout = copy_with_dev ? DW_DMA_LONG_TIMEOUT : DW_DMA_SHORT_TIMEOUT;
```

Comment:
Looking deeper, we could keep the single timeout and say

1) host timeout = DW_DMA_TIMEOUT (100uS)
2) DAI timeout = period size + DW_DMA_TIMEOUT (i.e. period + 100uS)

What do you think ? Simplifies further and means we are not restricted to 1ms periods.

Reply:
Idea to use period size is ok, I introduce it in code, but I'm not sure about timeout value bigger than period size.
When we stand that processing time for whole topology should be up to period value (1ms), and agent will panic when this time grow to 2x period (2ms), then with timeout value bigger than period (1ms) we lead to firmware crash instead of trace warning about timeout -> 1ms for processing + 1.1 ms timeout > 2ms

Reply:
Simplest solution will be to change agent timeout value above 2x period (2.5 for example), then timeout up to one period will be ok.
The most complicated is creation of separate component state STOP_BEGIN - quite dirty.
Another one is to separate PAUSE and STOP, because in PAUSE there is no need to wait until FIFO will be empty (because we just stop data draining, DMA is still active).
I think that last option sounds best for me.

Reply:
Yep, lets separate pause and stop. If DAI timeout is just for pause and stop then we can have it slightly bigger than remaining data in period (e.g. we can read SAR/DAR). This would guarantee our timeout is based on remaining data in current LLI only.